### PR TITLE
Create vertical layout option

### PR DESCRIPTION
**Pull request by mhkeller**
https://github.com/vsergeev/briefsky/issues/9 Creates an option in settings to have a vertical layout, [similar to Dark Sky](https://nightingaledvs.com/dark-sky-weather-data-viz/)

Here's a first go at it. Needs more mobile testing and check if the math is right aligning the times alongside the vertical bar and that the temperatures are scaled correctly.

Let me know what you think.

Screen Shot 2023-04-08 at 1 15 10 AM

### CODE REVIEW

**Comment by mhkeller** [comment]
I'm playing around with some details for aligning the weather timestamps to the vertical bar. In Dark Sky, it appears that the timestamp falls near the middle of that bar. So hours `8am` and `10am` are labeled and rain starts at `9am` the blue bar will start halfway between `8am` and `10am`. Setting a height of `100 / 12` sets the label a little differently. I'm trying to get it to align center without throwing off the first label... Work in progress...

**Comment by mhkeller** [comment]
@vsergeev I think it's ready for you to take a look. Let me know what you think:
Screen Shot 2023-04-08 at 4 35 34 PM

**Comment by vsergeev** [comment]
Yeah, I think a relative view height would be good in place of the fixed height.

**Comment by mhkeller** [comment]
Some updates:

> The temperature pills aren't centered with the line for me

This should be fixed. 

>  I'd like to avoid any fixed pixel heights on containers (e.g. height: 525px for the container in HourlyDetails)

Fixed. Removing the height seemed to work fine.

> Commit 9679c95 appears to remove text colors and drop shadows, which breaks the formatting on the horizontal view. (Why were these deleted?)

Fixed

> No need to commit the production bundle -- I can do that on an actual release with a version number change

Fixed

> The commits can also be squashed down into two commits, one that adds the layout setting (first commit), and one that adds the vertical layout.

Fixed

>  I think it would look better at 80% width and centered

This is the only thing that didn't work out well.The width becomes too narrow for there to be much variation on the temperature scale. Here's a screenshot. 
IMG_2964

Separately:

I added the sunrise and sunset times back and I standardized their look between the two components.

**Comment by mhkeller** [comment]
I increased the padding in some places for better tap accessibility and lightened up the temperature range so it's more easily scannable to the eye.

**Comment by mhkeller** [comment]
One other thing that's missing from today's view is when it's raining, it should display the Precipitation and Amount fields.

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "version": "1.0.0",
   "type": "module",
   "scripts": {
-    "dev": "vite",
+    "dev": "vite --host",
```
Comment (vsergeev) [inline package.json:7]: This doesn't belong in this commit or PR.

File: src/components/DailyPrecipitationDetails.svelte [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  import type { DailyPrecipitation } from '../providers/Provider';
+
+  import Amount from './scalars/Amount.svelte';
+
+  export let precipitation: DailyPrecipitation;
+</script>
+
+{#if precipitation.precipitation_probability !== undefined && precipitation.precipitation_amount !== undefined}
+	<div>
+		<span class="text-sm md:text-base font-semibold">Precipitation:</span> <span class="text-sm md:text-base">{precipitation.precipitation_probability}%</span>
```
Comment (vsergeev) [inline src/components/DailyPrecipitationDetails.svelte:11]: These line indents need to be normalized. You can run `npm run format` to format automatically.

File: src/components/scalars/TemperatureRange.svelte [hunk 3]
```diff
@@ -19,7 +19,7 @@
   <div class="text-right min-w-fit" style="min-width: {margin_left === 0 ? 'min-content' : `${margin_left}%`};">
     <Temperature value={low} />
   </div>
-  <div class="bg-gray-800 dark:bg-gray-400 my-0.5 mx-1.5 rounded-xl" style="width: {100 - margin_right - margin_left}%;" />
+  <div class="bg-gray-400 dark:bg-gray-400 my-0.5 mx-1.5 rounded-xl" style="width: {100 - margin_right - margin_left}%;" />
```
Comment (vsergeev) [inline src/components/scalars/TemperatureRange.svelte:22]: This change affects the default color scheme...

File: src/providers/Provider.ts [hunk 4]
```diff
@@ -74,6 +74,15 @@ export interface ProviderFactory {
   fromParams(params: object, location?: Location): Provider | null;
 }
 
+export interface SunTimes {
```
Comment (vsergeev) [inline src/providers/Provider.ts:77]: Types here should be limited to the boundary between the backend provider and the frontend, not auxiliary types for components. I think those components are simple enough that they aren't really necessary, anyway.

**Review by vsergeev — CHANGES_REQUESTED** [review]
This PR introduces unrelated changes and breaks the default color scheme and formatting. I think it's headed in the right direction overall, but I have to emphasize that the PR should have **absolutely no** impact on the rest of the app or to the existing horizontal layout. It also shouldn't try to incorporate new features that are outside the scope of the PR ("Create vertical layout option").

High level feedback:

* Adding precipitation and sunrise/sunset information to the top details should be a separate discussion and possibly PR. In it's current form, I'm not willing to add those fields because they're poorly formatted on desktop with two line rows that impair the readability of the other fields, and it's an open question whether these fields here add to the app or compromise its simplicity.
* For the width, we can use a media query to be 100% in mobile and 75% on desktop, so it's easy to read the vertical layout on both (without excessive left-right eye scanning on desktop).
* The temperature pills are vertically aligned with the lines, but I think it would look better if the hour and description on the left were vertically aligned with the lines too.
* Please compare the effect of your branch with the current deployment in both mobile and desktop. There are some regressions -- I've attached two screenshots as examples:

Original:
before

This Branch:
after

File: package.json (hunk 1 above)
Comment (mhkeller) [reply package.json:7]: This is just to help test and can be removed before the final merge

File: src/components/scalars/TemperatureRange.svelte (hunk 3 above)
Comment (mhkeller) [reply src/components/scalars/TemperatureRange.svelte:22]: I'll put this in a separate PR

**Comment by mhkeller** [comment]
Sure thing. I incorporated some accessibility and duplicate content fixes that I encountered but can create new issues for those. I can explain the reasoning for them in those issues and would be in favor of including them since there is some duplicate information that would adversely affect the UX for the vertical layout.

**Comment by mhkeller** [comment]
> The temperature pills are vertically aligned with the lines, but I think it would look better if the hour and description on the left were vertically aligned with the lines too.

Can you explain what you mean by this? Do you mean the horizontal line looks like it's not at the centerline of the text? Maybe you can send a screenshot of what you're seeing so I can make sure the fix is getting at the right thing.

**Comment by vsergeev** [comment]
> > The temperature pills are vertically aligned with the lines, but I think it would look better if the hour and description on the left were vertically aligned with the lines too.
> 
> Can you explain what you mean by this? Do you mean the horizontal line looks like it's not at the centerline of the text? Maybe you can send a screenshot of what you're seeing so I can make sure the fix is getting at the right thing.

I think so. Sure, here is a screenshot:

alignment

**Comment by mhkeller** [comment]
Got it. Yea that looks different from mine. What browser and OS are you using so I can try to replicate?

**Comment by vsergeev** [comment]
> Got it. Yea that looks different from mine. What browser and OS are you using so I can try to replicate?

I'm using Firefox 111 on Linux. I see it in Google Chrome 111 too.

**Comment by vsergeev** [comment]
I've added a View tab to the Settings on `master`, which will be a better place for a layout option like this.

**Comment by mhkeller** [comment]
@msft-jeelpatel, I didn't have the bandwidth to test this out since it was fine on my machine and I don't a Linux to test on. I won't have the time to finish this PR but if someone wants to pick it up, feel free to! I'll close this and if someone wants to resurrect it, then by all means they should

**Comment by mhkeller** [comment]
Reopening if other ppl want to work on it

**Comment by patel-jeel92** [comment]
> Reopening if other ppl want to work on it

I wish I knew how to fix some of the layout issues :/. 

For example, the "Now" starts at the very top of the weather bar 
image

Compared to DarkSky it looked like there was equal padding at the top and bottom. Currently, since the "Now" starts at the very top margin of the weather bar, there is quite a lot of unused space at the bottom

image

Other than that, I can try taking this forward

**Comment by patel-jeel92** [comment]
Hi @vsergeev I have started a new PR #21 to add the vertical layout. The reason i chose to add a new PR is because i have forked the repo and created a new branch off of it. Plus this had a bunch on conflicts and i found it easier to start over.

Huge thanks to @mhkeller for starting the initial work!

**Pull request closed without merging**